src/main/java/com/devathon/preguntonic/config/WebSocketConfig.java

```diff
@@ -0,0 +1,53 @@
+/*
+ *
+ * Copyright (c) 2024 Devathon., All Rights Reserved.
+ *
+ */
+package com.devathon.preguntonic.config;
+
+import org.springframework.context.annotation.Configuration;
+import org.springframework.messaging.simp.config.MessageBrokerRegistry;
+import org.springframework.web.socket.config.annotation.EnableWebSocketMessageBroker;
+import org.springframework.web.socket.config.annotation.StompEndpointRegistry;
+import org.springframework.web.socket.config.annotation.WebSocketMessageBrokerConfigurer;
+
+/** This file contains the configuration for the WebSocket. */
+@Configuration
+@EnableWebSocketMessageBroker
+public class WebSocketConfig implements WebSocketMessageBrokerConfigurer {
+
+  /**
+   * This method registers the endpoint for the WebSocket.
+   *
+   * @param registry the registry of the Stomp endpoints.
+   */
+  @Override
+  public void registerStompEndpoints(StompEndpointRegistry registry) {
+    registry
+        .addEndpoint("/preguntonic") // The endpoint for the WebSocket.
+        .setAllowedOriginPatterns("*") // The allowed origins.
+        .withSockJS(); // Enable SockJS -> Add support to clients that don't support WebSocket.
```

Comment:
👍  🚀 
Question about this 2 lines:

1.  when we move to "Production" the `setAllowedOriginPatterns` should it be removed?
2. Meanwhile I'm trying to understand better the `.withSockJS()` line I found this [spring-boot-websocket-without-sockjs](https://stackoverflow.com/questions/67920378/spring-boot-websocket-without-sockjs/68076180#68076180) question looks like it is for support `http` in the connection for the clients and without this line you can connect using `ws`. So do you think we need to add support for the other connection (`ws`)?
3. Just for curiosity do you know how to enable secure protocole? (`https`, `wss`)

Reply:
1. Should be replaced by our own domain (example.com). Althought, if we want to open connection to internet, we can keep this line
2. Yes! We can add web socket support and keep http support, so i will add web socket support.
3. We cant add secure protocols from backend code, we need to deploy on a remote server and install lets encrypt certificates